wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/Data.kt

```diff
@@ -1,9 +1,25 @@
 package org.stellar.walletsdk.auth
 
+import kotlinx.serialization.Contextual
 import kotlinx.serialization.Serializable
+import org.stellar.walletsdk.STRING_TRIM_LENGTH
 
-@Serializable data class ChallengeResponse(val transaction: String, val network_passphrase: String)
+@Suppress("ConstructorParameterNaming")
+@Serializable
+data class ChallengeResponse(val transaction: String, val network_passphrase: String)
 
-@Serializable internal data class AuthToken(val token: String)
+@Serializable
+@JvmInline
+value class AuthTokenValue(private val value: String) {
+  fun prettify(): String {
+    return value.take(STRING_TRIM_LENGTH)
+  }
+
+  override fun toString(): String {
+    return value
+  }
+}
+
+@Serializable internal data class AuthToken(@Contextual val token: AuthTokenValue)
```

Comment:
I'm not sure if you even need this class now

Reply:
Can we use `AuthToken` both for value and response for the auth which returns `token`?

Reply:
If it returns object, you may still need this data class wrapper (or you can make a custom deserializer) 
Probably easier to just leave a wrapper, but please rename AuthTokenValue -> AuthToken and AuthToken -> AuthTokenResponse